packages/graph/hash_graph/lib/graph/src/api/rest/mod.rs

```diff
@@ -109,3 +144,46 @@ impl Modify for MergeAddon {
         );
     }
 }
+
+/// Addon to allow external references in schemas.
+///
+/// Any component that starts with `EXTERNAL_` will transform into a relative URL in the schema and
+/// receive a `.json` ending.
+///
+/// For example the `EXTERNAL_Entity` component will be transformed into `./models/Entity.json`
```

Comment:
Why is it that we're mixing up casings? I understand wanting an identifiable prefix but the filenames and special-words are really ugly, especially a mix of all-caps, snake case, and pascal-case 😁 

Do we perhaps just want `external_data_type` rather than `EXTERNAL_DataType`? Or perhaps we want to go all-caps with `EXTERNAL_DATA_TYPE` or something.

Reply:
I would have really liked if the `body` attribute supported arbitrary terms, then I'd have just put a dot instead of an underscore. The reason what just to make it clear that it's not a part of the components. I would be fine with `external_data_type`, which means having `data_type.json` in the json_schemas folder too

Reply:
I'm assuming we can't use `$data_type` or something either?

Reply:
I think the all-caps might be better to identify it as some variable/special thing. We can just `to_lowercase` in the impl and keep the file names nice

Reply:
Addressed in ec96248ffc18bbcefff37085c9175e465b4faddb